File: src/main/java/domain/stock/History.java
package domain.stock;

//===========================================================//
/**
 * Represents the historical data of a stock for a single trading day.
 *
 * @param closingPrice - the closing price of the asset.
 */
//===========================================================//

public record History(
    double closingPrice
) {
    public History {
        if(closingPrice <= 0d) throw new IllegalArgumentException("ClosingPrice");
    }
}

File: src/main/kotlin/domain/stock/History.kt
package domain.stock

//===========================================================//
/**
 * Represents the historical data of a stock for a single trading day.
 * 
 * @param closingPrice - the closing price of the asset.
 */
//===========================================================//

data class History(
    val closingPrice: Double
) {
    init {
        require(closingPrice >= 0.0) { "ClosingPrice" }
    }
}
